src/components/pastAnalysis/PastAnalysisTable.tsx

```diff
@@ -43,37 +50,125 @@ const PastAnalysisTable = () => {
   ]
   const [rows, setRows] = useState<(tableRowsChild | tableRowsParent)[]>([])
 
+  // Stores properties of the table used for pagination
+  const [tableStates, setTableStates] = useState<TableStates>({
+    page: 0, // Current table page number
+    maxFeedId: -1, // ID of the latest Feed on Swift as of when PastAnalysisTable first mounted OR was last reset
+    lastOffset: 0, // Page offset value for where to begin fetching the next unseen page
+    lastPage: -1, // Table page number of the very last page (-1 means last page has not yet been seen)
+    storedPages: [] // Stores pages that have been seen in an array of pages
+  });
+
+  // Stores an array of the "Analysis Created" property of the rows of page 0 of the table
+  // Used to identify which rows are new and need to be highlighted green
+  const [newRowsRef, setNewRowsRef] = useState<string[]>([]);
+
+  // Reset table and update the maxFeedId to the latest Feed ID in Swift
+  const updateMaxFeedId = () => {
+    // Right before resetting, get a list of all the "Analysis Created" properties on the on page 0
+    setNewRowsRef(tableStates.storedPages[0]?.map((study: StudyInstanceWithSeries) => study.analysisCreated));
+
+    ChrisIntegration.getLatestFeedId().then((id: number) => {
+      setTableStates({
+        page: 0,
+        maxFeedId: id,
+        lastOffset: 0,
+        lastPage: -1,
+        storedPages: []
+      })
+    })
+  }
+
+  // Reset table and update the maxFeedId when the PastAnalysisTable first mounts
+  useEffect(() => {
+    updateMaxFeedId();
+  }, []);
+
+  // If new past analyses are available, reset table to initial state and update maxFeedId
+  useEffect(() => {
+    if (areNewImgsAvailable) {
+      setLoading(true);
+      updateMaxFeedId();
```

Comment:
How come we need to call this method each time new images become available?

Reply:
The `maxFeedId` represents the latest Feed ID that was seen on Swift the last time PastAnalysisTable rendered. Whenever there are new images available (AKA analyses are done processing), the desired behaviour is to have the table "refresh", which means the table resets itself and begins fetching rows from offset 0 on Swift.